### PR TITLE
ResNet50:add resnet50 pytorch case for Nvidia

### PR DESCRIPTION
**Pull request by yuzhou03**
1. add resnet50 pytorch case
2. add nvidia configs
3. running with specified accelerators support

### CODE REVIEW

File: training/run_benchmarks/pytorch/start_pytorch_task.py [hunk 1]
```diff
@@ -117,9 +121,12 @@ def _set_common_ddp_envs(task_args):
 
     # set GPU/MLU device env, TODO other vendor's device
     if task_args.visible_dev_env is not None:
-        acce_visible = range(0, task_args.nproc)
-        current_env[task_args.visible_dev_env] = ",".join(
-            str(_id) for _id in acce_visible)
+        # acce_visible = range(0, task_args.nproc)
```
Comment (yuzhou03) [inline training/run_benchmarks/pytorch/start_pytorch_task.py:124]: revert

File: training/benchmarks/resnet50/pytorch/model/models/modeling.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                # available GPUs if device_ids are not set
+                # model = torch.nn.parallel.DistributedDataParallel(model)
+    elif args.gpu is not None and torch.cuda.is_available():
+        torch.cuda.set_device(args.gpu)
+        model = model.cuda(args.gpu)
+    elif hasattr(torch.backends, "mps") and torch.backends.mps.is_available():
+        device = torch.device("mps")
+    else:
+        # DataParallel will divide and allocate batch_size to all available GPUs
+        if args.arch.startswith('alexnet') or args.arch.startswith('vgg'):
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/model/models/modeling.py:42]: remove

File: training/benchmarks/resnet50/pytorch/run_pretraining.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+    # train && val dataset
+    train_dataset = build_train_dataset(config)
+    val_dataset = build_eval_dataset(config)
+    train_dataloader = build_train_dataloader(config,
+                                              train_dataset,
+                                              worker_init_fn=None)
+    eval_dataloader = build_eval_dataloader(config, val_dataset)
+
+    if config.debug_mode:
```
Comment (yuzhou03) [inline training/benchmarks/resnet50/pytorch/run_pretraining.py:72]: config_manager.py中，输出所有生效的配置。

File: .gitignore [hunk 4]
```diff
@@ -3,4 +3,10 @@
...
 __pycache__/
-.pytest_cache
\ No newline at end of file
+.pytest_cache
+
+
+training/result/
+
+*.tar
+training/run_benchmarks/rerun.sh
```
Comment (upvenly) [inline .gitignore:12]: 个人调试的代码不要提交上来

File: training/benchmarks/resnet50/pytorch/config/_base.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
+# chip vendor, required
+vendor: str = None
+
+# random seed
+seed: int = 1234
+
+# model args
+arch: str = "resnet50"
+# model name
+name: str = "resnet50"
```
Comment (upvenly) [inline training/benchmarks/resnet50/pytorch/config/_base.py:10]: 可以复用arch 这个参数吗，不必要写两次

File: training/benchmarks/resnet50/pytorch/config/_base.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+# Minimum loss scale for dynamic loss scale
+min_scale: float = 1
+
+# distributed args
+# Turn ON gradient_as_bucket_view optimization in native DDP.
+use_gradient_as_bucket_view: bool = False
+
+# load and save args
+# Path to a directory containing a model checkpoint.
+init_checkpoint: str = None
```
Comment (upvenly) [inline training/benchmarks/resnet50/pytorch/config/_base.py:49]: resnet50 每次从头开始训吗

File: training/benchmarks/resnet50/pytorch/model/models/modeling.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        # should always set the single device scope, otherwise,
+        # DistributedDataParallel will use all available devices.
+        if torch.cuda.is_available():
+            if args.gpu is not None and args.gpu != 0:
+                torch.cuda.set_device(args.gpu)
+                model.cuda(args.gpu)
+                # When using a single GPU per process and per
+                # DistributedDataParallel, we need to divide the batch size
+                # ourselves based on the total number of GPUs of the current node.
+                args.train_batch_size = int(args.train_batch_size / ngpus_per_node)
```
Comment (upvenly) [inline training/benchmarks/resnet50/pytorch/model/models/modeling.py:27]: args.train_batch_size 定义为单卡bs，不必要除以 ngpus_per_node

File: training/benchmarks/resnet50/pytorch/model/models/modeling.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+                model = torch.nn.parallel.DistributedDataParallel(model, device_ids=[args.gpu])
+            else:
+                model.cuda()
+                # DistributedDataParallel will divide and allocate batch_size to all
+                # available GPUs if device_ids are not set
+                # model = torch.nn.parallel.DistributedDataParallel(model)
+    elif args.gpu is not None and torch.cuda.is_available():
+        torch.cuda.set_device(args.gpu)
+        model = model.cuda(args.gpu)
+    elif hasattr(torch.backends, "mps") and torch.backends.mps.is_available():
```
Comment (upvenly) [inline training/benchmarks/resnet50/pytorch/model/models/modeling.py:38]: 我们不支持mps，这部分判断不必要保留

File: training/benchmarks/resnet50/pytorch/run_pretraining.py [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    dist_pytorch.init_dist_training_env(config)
+    dist_pytorch.barrier(config.vendor)
+    model_driver.event(Event.INIT_START)
+
+    # logger
+    logger = model_driver.logger
+    init_start_time = logger.previous_log_time  # init起始时间，单位ms
+
+    # TODO 得到seed
+    seed = config.seed if config.seed else 0
```
Comment (upvenly) [inline training/benchmarks/resnet50/pytorch/run_pretraining.py:56]: 默认值给1024 吧，一般不给0

File: training/benchmarks/resnet50/pytorch/run_pretraining.py [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+        adapter=trainer_adapter,
+        evaluator=evaluator,
+        training_state=training_state,
+        device=device,
+        config=config,
+    )
+    training_state.set_trainer(trainer)
+
+    # 设置分布式环境
+    dist_pytorch.barrier(config.vendor)
```
Comment (upvenly) [inline training/benchmarks/resnet50/pytorch/run_pretraining.py:91]: 这里为啥要两次barrier 啊

File: training/benchmarks/resnet50/pytorch/run_pretraining.py [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+    trainer.init()
+    dist_pytorch.barrier(config.vendor)
```
Comment (upvenly) [inline training/benchmarks/resnet50/pytorch/run_pretraining.py:93]: 同上

File: training/benchmarks/resnet50/pytorch/run_pretraining.py [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+                                     raw_train_start_time) / 1e3
+    return config, training_state
+
+
+if __name__ == "__main__":
+    start = time.time()
+    config, state = main()
+    if not dist_pytorch.is_main_process():
+        sys.exit(0)
+    finished_info = get_finished_info(start, state, config.do_train,
```
Comment (upvenly) [inline training/benchmarks/resnet50/pytorch/run_pretraining.py:149]: 换成非调用的吧，我们封装的方法直接删除。另外，需要在case添加规范里增加对这部分的规范要求，包括不限于 finished_info 需要提供哪些关键信息

File: training/benchmarks/resnet50/pytorch/train/device.py [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+    
+    @staticmethod
+    def get_device(args):
+        device = None
+        if torch.cuda.is_available():
+            if args.gpu:
+                device = torch.device('cuda:{}'.format(args.gpu))
+            else:
+                device = torch.device("cuda")
+        elif hasattr(torch.backends, "mps")  and torch.backends.mps.is_available():
```
Comment (upvenly) [inline training/benchmarks/resnet50/pytorch/train/device.py:13]: mps 不必要

File: training/benchmarks/resnet50/pytorch/train/meter.py [hunk 14]
```diff
@@ -0,0 +1,103 @@
+from enum import Enum
```
Comment (upvenly) [inline training/benchmarks/resnet50/pytorch/train/meter.py:1]: 我们的log 或者计算平均体系可以参考这个文件重构

File: training/benchmarks/resnet50/pytorch/train/training_state.py [hunk 15]
```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass
```
Comment (upvenly) [inline training/benchmarks/resnet50/pytorch/train/training_state.py:1]: 这个类基本不变，有没有可能抽象出来，做成基类，模型只继承

File: training/nvidia/resnet50-pytorch/config/config_A100x1x4.py [hunk 16]
```diff
@@ -0,0 +1,23 @@
+#from extern.training_even import ApexTrainingEvent
+from config_common import *
+
+fp16 = True
+dist_backend = "nccl"
+target_embedding_average = 0.92
+
+gradient_accumulation_steps = 1
+
+train_batch_size = 256
```
Comment (upvenly) [inline training/nvidia/resnet50-pytorch/config/config_A100x1x4.py:10]: 改成单卡的bs

File: training/nvidia/resnet50-pytorch/config/config_A100x1x2.py [hunk 17]
```diff
@@ -0,0 +1,23 @@
+#from extern.training_even import ApexTrainingEvent
+from config_common import *
+
+fp16 = True
+dist_backend = "nccl"
+target_embedding_average = 0.92
+
+gradient_accumulation_steps = 1
+
+train_batch_size = 256
```
Comment (upvenly) [inline training/nvidia/resnet50-pytorch/config/config_A100x1x2.py:10]: 改成单卡bs

File: training/nvidia/resnet50-pytorch/config/config_A100x1x4.py [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+eval_batch_size = train_batch_size
+max_steps = 10000000
+max_samples_termination = 43912600
+
+warmup = 0.2
+learning_rate = 0.01
```
Comment (upvenly) [inline training/nvidia/resnet50-pytorch/config/config_A100x1x4.py:16]: 不同的config之间有什么区别吗，没有区别的话就不必要写几遍

File: training/nvidia/resnet50-pytorch/config/config_A100x1x8.py [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+
+warmup = 0.2
+learning_rate = 0.01
+
+beta_1: float = 0.9
+beta_2: float = 0.99
+eps: float = 1e-08
+
+seed = 23333
+training_event = None
```
Comment (upvenly) [inline training/nvidia/resnet50-pytorch/config/config_A100x1x8.py:23]: 不同的config之间有什么区别吗，没有区别的话就不必要写几遍

File: training/nvidia/resnet50-pytorch/config/config_A100x2x8.py [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+warmup = 0.2
+learning_rate = 0.01
+
+beta_1: float = 0.9
+beta_2: float = 0.99
+eps: float = 1e-08
+
+seed = 23333
+max_samples_termination = 43912600
+training_event = None
```
Comment (upvenly) [inline training/nvidia/resnet50-pytorch/config/config_A100x2x8.py:24]: 不同的config之间有什么区别吗，没有区别的话就不必要写几遍

File: training/nvidia/resnet50-pytorch/config/requirements.txt [hunk 21]
```diff
@@ -0,0 +1 @@
+numpy>=1.15.4
```
Comment (upvenly) [inline training/nvidia/resnet50-pytorch/config/requirements.txt:1]: 这个是必须的吗

File: training/run_benchmarks/config/cluster_conf.py [hunk 22]
```diff
@@ -1,7 +1,7 @@
 '''Cluster configs'''
 
 # Hosts to run the benchmark. Each item is an IP address or a hostname.
-HOSTS = ["10.1.2.2", "10.1.2.3", "10.1.2.4"]
+HOSTS = ["10.1.2.155", "10.1.2.158"]
```
Comment (upvenly) [inline training/run_benchmarks/config/cluster_conf.py:4]: 提交的代码不要带有公司内部实际IP

File: training/run_benchmarks/config/test_conf.py [hunk 23]
```diff
@@ -18,7 +18,7 @@
 
 # The path that flagperf deploy in the cluster.
 # If not set, it will be os.path.dirname(run.py)/../../training/
-FLAGPERF_PATH_HOST = "/home/flagperf/training"
+FLAGPERF_PATH_HOST = "/home/zhouyu/workspace/FlagPerf/training"
```
Comment (upvenly) [inline training/run_benchmarks/config/test_conf.py:21]: 提交的代码不要带有个人目录

File: training/run_benchmarks/config/test_conf.py [hunk 24]
```diff
@@ -39,8 +39,17 @@
...
+    # 'BERT_PADDLE_DEMO_A100_1X4',
+    # 'BERT_PADDLE_DEMO_A100_1X2',
+    # 'BERT_PADDLE_DEMO_A100_1X1',
+    # 'GLM_TORCH_DEMO_A100_1X8',
+    # 'CPM_TORCH_DEMO_A100_1X8'
+    'RESNET50_TORCH_DEMO_A100_2X8',
+    'RESNET50_TORCH_DEMO_A100_1X8',
+    'RESNET50_TORCH_DEMO_A100_1X4',
+    'RESNET50_TORCH_DEMO_A100_1X2',
+    'RESNET50_TORCH_DEMO_A100_1X1',
```
Comment (upvenly) [inline training/run_benchmarks/config/test_conf.py:52]: 我们需要约定一个提交的test_conf 的格式

File: training/run_benchmarks/config/test_conf.py [hunk 25]
```diff
@@ -88,3 +97,58 @@
...
+RESNET50_TORCH_DEMO_A100_2X8 = {
+    "model": "resnet50",
+    "framework": "pytorch",
+    "config": "config_A100x1x8",
+    "repeat": 1,
+    "nnodes": 2,
+    "nproc": 8,
+    "data_dir_host": "/raid/dataset/ImageNet/imagenet",
+    "data_dir_container": "/mnt/data/resnet50/train/",
+}
```
Comment (upvenly) [inline training/run_benchmarks/config/test_conf.py:154]: 基于上周五会议意见修改

File: training/run_benchmarks/run.py [hunk 26]
```diff
@@ -545,7 +545,7 @@ def main():
                     + " --log_dir " + log_dir_container \
                     + " --log_level " + tc.FLAGPERF_LOG_LEVEL \
                     + " --extern_config_file " + case_config["config"] \
-                    + ".py" + " --enable_extern_config "
+                    + ".py" + " --enable_extern_config " + tc.ACCE_CONTAINER_OPT
```
Comment (upvenly) [inline training/run_benchmarks/run.py:548]: 这个参数的作用是？

File: training/nvidia/resnet50-pytorch/config/requirements.txt (hunk 21 above)
Comment (yuzhou03) [reply training/nvidia/resnet50-pytorch/config/requirements.txt:1]: dataloader中使用了np.random.seed

File: training/benchmarks/resnet50/pytorch/run_pretraining.py (hunk 10 above)
Comment (yuzhou03) [reply training/benchmarks/resnet50/pytorch/run_pretraining.py:91]: trainer_init()方法前后，各需要一次barrier. 参考CPM和GLM

File: training/run_benchmarks/config/model_config/resnet50.py
Comment (yuzhou03) [inline training/run_benchmarks/config/model_config/resnet50.py:1]: revert to test_conf.py
